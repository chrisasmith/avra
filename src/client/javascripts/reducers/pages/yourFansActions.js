import { api } from '../../services'
import { createRequests, doApiCall, createRequestTypes } from '../actionHelper'

export const GET_HOME_PAGE = createRequestTypes('GET_HOME_PAGE')
export const getHomePage = createRequests(GET_HOME_PAGE)

export const HOME_PAGE_DATA_SUCCESS = 'HOME_PAGE_DATA_SUCCESS'
export function homePageDataSuccess(data) {
  return { type: HOME_PAGE_DATA_SUCCESS, data }
}

export function getHomeData() {
  return doApiCall({
    api: api.getHomeData,
    action: getHomePage,
  })
}
