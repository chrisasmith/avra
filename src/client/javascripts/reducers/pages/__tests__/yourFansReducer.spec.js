import reducer from '../yourFansReducer'
import * as ActionTypes from '../yourFansActions'

describe('Your Fans reducer', () => {
  it('should handle GET_HOME_PAGE.SUCCESS', () => {
    expect(
      reducer({}, {
        type: ActionTypes.GET_HOME_PAGE.SUCCESS,
        response: {
          data: [],
        },
      })
    ).toMatchSnapshot()
  })
})
