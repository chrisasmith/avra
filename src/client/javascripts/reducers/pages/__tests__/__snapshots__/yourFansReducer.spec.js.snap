// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Your Fans reducer should handle GET_HOME_PAGE.SUCCESS 1`] = `Object {}`;
